### PR TITLE
contrib/IBM/sarama: adds trace for IBM/sarama

### PR DESCRIPTION
**Pull request by rodrigoff**
### What does this PR do?

As of version 1.40.0 and moving forward, sarama's ownership has been transfered from Shopify to IBM.

This change adds the contrib package for the new package as a copy of contrib/Shopify/sarama (in order not to break anything) to allow for apps to migrate to the new package.

### Motivation

See https://github.com/IBM/sarama/issues/2461 for more details

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: contrib/IBM/sarama/sarama.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (knusbaum) [inline contrib/IBM/sarama/sarama.go:1]: The path for this needs to be `contrib/IBM/sarama.v1/`

File: contrib/IBM/sarama/sarama.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+	"github.com/IBM/sarama"
+)
+
+const componentName = "IBM/sarama"
+
+func init() {
+	telemetry.LoadIntegration("IBM/sarama")
```
Comment (knusbaum) [inline contrib/IBM/sarama/sarama.go:24]: We want to add a `tracer.MarkIntegrationImported("github.com/IBM/sarama.v1")` line here like we do in the current `sarama` package.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Thanks for the contribution. 
There are just a few minor things to take care of.

File: go.mod [hunk 3]
```diff
@@ -12,7 +12,8 @@ require (
 	github.com/DataDog/go-libddwaf v1.4.1
 	github.com/DataDog/gostackparse v0.5.0
 	github.com/DataDog/sketches-go v1.2.1
-	github.com/Shopify/sarama v1.22.0
+	github.com/IBM/sarama v1.40.0
+	github.com/Shopify/sarama v1.19.0
```
[the remark this answers is not in the document]
Comment (rodrigoff) [reply go.mod]: No reason, that was unintentional.

File: contrib/IBM/sarama/sarama.go (hunk 1 above)
Comment (rodrigoff) [reply contrib/IBM/sarama/sarama.go]: I've updated the path.

**Review by hannahkm — CHANGES_REQUESTED** [review]

File: contrib/IBM/sarama.v1/sarama.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+	"github.com/IBM/sarama"
+)
+
+const componentName = "IBM/sarama"
+
+func init() {
+	telemetry.LoadIntegration("IBM/sarama")
```
[the remark this answers is not in the document]
Comment (rodrigoff) [reply contrib/IBM/sarama.v1/sarama.go]: Updated.

File: ddtrace/tracer/option.go [hunk 5]
```diff
@@ -85,6 +85,7 @@ var contribIntegrations = map[string]struct {
 	"gopkg.in/olivere/elastic.v3":                   {"Elasticsearch v3", false},
 	"github.com/redis/go-redis/v9":                  {"Redis v9", false},
 	"github.com/segmentio/kafka-go":                 {"Kafka v0", false},
+	"github.com/IBM/sarama":                         {"Kafka (sarama)", false},
 	"github.com/Shopify/sarama":                     {"Kafka (sarama)", false},
```
[the remark this answers is not in the document]
Comment (rodrigoff) [reply ddtrace/tracer/option.go]: I agree, I was just trying to keep consistency with `confluent-kafka-go` which is named `Kafka (confluent)` (and `Kafka (confluent) v2`, but looking at other `contrib` names, I don't think that's necessary.

I've updated both names to better represent their integration.

**Review by zarirhamza — APPROVED** [review]
LGTM, just added a quick fix to the tracer test where we have a count of all existing integrations.

**Review by knusbaum — APPROVED** [review]
Thanks for working with us on this! 
Looks great!

**Pull request merged**